File: runtime/android/java/src/org/xwalk/core/XWalkFormDatabase.java
// Copyright 2013 The Chromium Authors. All rights reserved.
// Copyright (c) 2013 Intel Corporation. All rights reserved. 
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.xwalk.core;

import org.chromium.base.JNINamespace;

/**
 * Exposes a subset of Chromium form database to xwalk database for managing autocomplete
 * functionality.
 */
@JNINamespace("xwalk")
public class XWalkFormDatabase {

    public static boolean hasFormData() {
        return nativeHasFormData();
    }

    public static void clearFormData() {
        nativeClearFormData();
    }

    //--------------------------------------------------------------------------------------------
    //  Native methods
    //--------------------------------------------------------------------------------------------
    private static native boolean nativeHasFormData();

    private static native void nativeClearFormData();
}

File: runtime/browser/android/xwalk_form_database.cc
// Copyright 2016 The Intel Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "xwalk/runtime/browser/android/xwalk_form_database.h"

#include "base/android/jni_android.h"

#include "xwalk/runtime/browser/xwalk_browser_context.h"
#include "jni/XWalkFormDatabase_jni.h"

namespace xwalk {

namespace {

XWalkFormDatabaseService* GetFormDatabaseService() {

  XWalkBrowserContext* context = XWalkBrowserContext::GetDefault();
  XWalkFormDatabaseService* service = context->GetFormDatabaseService();
  return service;
}

} // anonymous namespace

// static
jboolean HasFormData(JNIEnv*, const JavaParamRef<jclass>&) {
  return GetFormDatabaseService()->HasFormData();
}

// static
void ClearFormData(JNIEnv*, const JavaParamRef<jclass>&) {
  GetFormDatabaseService()->ClearFormData();
}

bool RegisterXWalkFormDatabase(JNIEnv* env) {
  return RegisterNativesImpl(env);
}

} // namespace xwalk
